### PR TITLE
Feature/ league stat tracker aw

### PR DESCRIPTION
**Pull request by andyweissman6**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] The branch created follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

* **What is the new behavior (if this is a feature change)?**

adds methods and tests for #count_of_teams, #best_offense, #worst_offense

* **Other information**:

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -82,4 +82,48 @@ def average_goals_by_season
...
+          game.away_goals.to_i
+        end
+      end.sum
+      team_avg_goals[team.team_name] = goals.to_f / team_games.count
+    end
+    team_avg_goals.max_by do |team_name, avg_goals|
+      avg_goals
+    end.first
+  end
+
```
Comment (AlexKiselich) [inline lib/stat_tracker.rb:109]: I used pretty much the same process for my methods. I'm sure we could refactor our methods to some helper methods to clean up our code. Maybe something we do if we have more time

File: lib/stat_tracker.rb
Comment (AlexKiselich) [inline lib/stat_tracker.rb:1]: Looks great! I'm glad a got to be there when you were going over this! Helped me out a bunch!!

File: spec/stat_tracker_spec.rb
Comment (AlexKiselich) [inline spec/stat_tracker_spec.rb:1]: Looks nice and clean, great job!

File: lib/stat_tracker.rb (hunk 1 above)
Comment (calforcal) [reply lib/stat_tracker.rb:109]: Definitely agree. With more time I think we find a helper method or two in many of these methods. Super good work though Andy.

**Pull request merged**